src/repository/storage/postgres/orm/sequelize/teams.ts

```diff
@@ -203,6 +217,57 @@ export default class TeamsSequelizeStorage {
     });
   }
 
+  /**
+   * Get note parent structure
+   * @param noteId : the ID of the note.
+   * @param userId : The ID of the user.
+   */
+  public async getAllNoteParents(noteId: NoteInternalId, userId: number): Promise<NoteParentStructure[]> {
+    if (!this.noteModel || !this.noteRelationModel) {
+      throw new DomainError('Note model or Note relation model not initialized');
+    }
+
+    const parentNotes: NoteParentStructure[] = [];
+    let currentNoteId: NoteInternalId | null = noteId;
+
+    while (currentNoteId != null) {
+      // Check if the user has access to the current note
+      const teamMember = await this.model.findOne({
+        where: {
+          noteId: currentNoteId,
+          userId,
+        },
+      });
+
+      if (teamMember) {
```

Comment:
you don't handle inheritance at all, current team can be empty, but inherited team would have user, and he would have access to the note